core/CronArchive.php

```diff
@@ -908,7 +908,7 @@ protected function processArchiveDays($idSite, $lastTimestampWebsiteProcessedDay
             $this->logArchiveWebsite($idSite, "day", $date);
 
             $content = $this->request($url);
-            $daysResponse = Common::safe_unserialize($content);
+            $daysResponse = json_decode($content, true);
```

Comment:
btw does it make sense to look here for json decode errors? Like before I remember I've seen quite a few times `'Unable to unserialize a string:` errors/exceptions when eg some other content was printed like warnings/errors or something else before or after the actual output

Reply:
I don't think json_decode will result in warning when the string is no valid json. At least for PHP 7. There's `json_last_error()` to get the error if there was any.

Reply:
`json_last_error ` that's what I meant 👍

Reply:
I'm not sure if it's worth to log a warning here if the request doesn't have a valid json response.
I guess the response might only not be valid if an error or warning is printed within, but that should already be logged within the request. Or what benefit would you see there?

Reply:
It's basically forwarding the error from the archiving request to the CronArchive. I've seen quite a few issues with this in the past 6 months and it was quite useful to fail the archiving here should there be any error.

Reply:
I see your point. But archiving will actually still fail because of the lines below:
https://github.com/matomo-org/matomo/blob/430a7f4c1f0cfceee7ede1d43c75e9e562fb718a/core/CronArchive.php#L910-L927

Reply:
Yes, you just don't know why maybe? A serialisation error be a bit more specific and can help investigate / understand things better

Reply:
Can't see a big difference between a (specific) serialization error and a `invalid response` error. But might make sense to make the one message already triggered a bit better...

Reply:
Done. See https://github.com/matomo-org/matomo/blob/b19ecc8d56201f4c1347a561208976c17acdf0f1/core/CronArchive.php#L923-L927